app/clients/ethereum/ethereum.go

```diff
@@ -131,3 +135,31 @@ func (ec *Client) waitForTransactionReceipt(hash common.Hash) (txReceipt *types.
 
 	return ec.Client.TransactionReceipt(context.Background(), hash)
 }
+
+func (ec *Client) WaitForConfirmations(raw types.Log) error {
+	target := raw.BlockNumber + ec.config.BlockConfirmations
+	for {
+		currentBlockNumber, err := ec.BlockNumber(context.Background())
+		if err != nil {
+			ec.logger.Errorf("[%s] Failed retrieving block number.", raw.TxHash.String())
+			return err
+		}
+
+		if target <= currentBlockNumber {
+			receipt, err := ec.TransactionReceipt(context.Background(), raw.TxHash)
+			if err != nil {
+				ec.logger.Infof("[%s] Ethereum TX went into an uncle block.", raw.TxHash.String())
+				return err
+			}
+
+			if receipt.Status == 1 {
+				ec.logger.Debugf("[%s] Transaction was successfully mined", raw.TxHash.String())
```

Comment:
nit: the transaction technically was already mined the moment we receive the event. Lets change the log to `[%] Transaction received [ec.config.BlockConfirmations] block confirmations`